docs/documentation/automation/bots/ccjs-manual.md

```diff
@@ -0,0 +1,46 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+<img alt="title image" className="img_sizing" src={useBaseUrl('img/design/Main_menu.svg')} />
+<br/>
+
+## Overview {#overview}
+
+With the _Custom Javascript Code_ stage-bot, you can add extra customization to your routines. This powerful tool runs safely in a sandboxed _node.js_ environment with the ability to extract, process, and return data for use in the rest of your routine.
+
+Simply add your Javascript code into the **source code** field in the stage-bot's settings panel:
+
+<img alt="ccjs settings panel" className="img_sizing item shadow--tl" src={useBaseUrl('img/automations_routines_stage_ccjs_00.png')} />
+<br/>
+
+### Default Variables {#default-variables}
+
+By default, your function can access two variables:
+
+- **Value** is an object that contains the [context](/docs/documentation/automation/triggers_and_contexts) data, this will depend on the [trigger](/docs/documentation/automation/triggers_and_contexts).
+- **Output** is an array of objects `{ key: 'step-name', result: Object }` with all the previous executed steps.
+
+**NOTE**: The code runs in a sandboxed _node.js_ environment. No _requires_ are allowed (or _will_ word) - so no storage or network requests will work.
+
+<div className="alert alert--secondary">
+
+### <span className="hero__subtitle">Examples:</span> {#example}
+
+One common example of how this tool can be used -but not limited to- is automated data extraction from _collections_ and returning the results in string format for use in the routine's upcoming stage.
+
+Another example -like the one shown below- has data taken from multiple stages and then processed to generate data for other steps.
```

Comment:
This example, takes a `list-input` value of "email"  or "name", to either convert  a list of `COTUsers` into a list of `emails` or `names`